src/client/datascience/dataScienceSurveyBanner.ts

```diff
@@ -77,21 +79,25 @@ export class DataScienceSurveyBannerLogger implements IInteractiveWindowListener
     }
 }
 
+export type ShowBannerWithExpiryTime = { data: boolean; expiry?: number };
 @injectable()
 export class DataScienceSurveyBanner implements IJupyterExtensionBanner {
     public get enabled(): boolean {
-        return (
-            this.persistentState.createGlobalPersistentState<boolean>(DSSurveyStateKeys.ShowBanner, true).value &&
-            env.uiKind !== UIKind?.Web &&
-            this.applicationEnvironment.channel === 'stable'
-        );
+        // if (this.applicationEnvironment.uiKind !== UIKind.Desktop || this.applicationEnvironment.channel !== 'stable') {
+        //     return false;
+        // }
+        if (!this.showBannerState.value.expiry) {
+            return true;
+        }
+        return this.showBannerState.value.expiry! < Date.now();
```

Comment:
Explicit check for expiry.
This is the structure of the data stored in persistent storage if it contains any expiry information (hence ensuring we keep using the old data).